packages/shell-api/src/shard.spec.ts

```diff
@@ -1217,6 +1217,275 @@ describe('Shard', () => {
+        serviceProvider.runCommandWithCheck.resolves({ ok: 1, msg: 'isdbgrid' });
+        const expectedError = new Error();
+        serviceProvider.updateOne.rejects(expectedError);
+        const caughtError = await shard.disableAutoMerger('ns')
+          .catch(e => e);
+        expect(caughtError).to.equal(expectedError);
+      });
+
+      it('throws if not mongos', async() => {
+        const expectedResult = { ok: 1 } as any;
+        serviceProvider.runCommandWithCheck.resolves({ ok: 1, msg: 'not dbgrid' });
+        serviceProvider.updateOne.resolves(expectedResult);
+        await shard.disableAutoMerger('ns');
+        expect(warnSpy.calledOnce).to.equal(true);
+      });
+    });
+
+    describe('enableAutoMerger', () => {
+      it('calls serviceProvider.updateOne', async() => {
+        serviceProvider.runCommandWithCheck.resolves({ ok: 1, msg: 'isdbgrid' });
+        const expectedResult = {
+          matchedCount: 1,
+          modifiedCount: 1,
+          upsertedCount: 1,
+          upsertedId: { _id: 0 },
+          result: { ok: 1, n: 1, nModified: 1 },
+          connection: null
+        } as any;
+        serviceProvider.updateOne.resolves(expectedResult);
+        await shard.enableAutoMerger('ns');
+
+        expect(serviceProvider.updateOne).to.have.been.calledWith(
+          'config',
+          'collections',
+          { _id: 'ns' },
+          { $unset: { enableAutoMerge: 1 } },
+          { writeConcern: { w: 'majority', wtimeout: 60000 } }
+        );
+      });
+
+      it('returns whatever serviceProvider.updateOne returns', async() => {
+        serviceProvider.runCommandWithCheck.resolves({ ok: 1, msg: 'isdbgrid' });
+        const oid = new bson.ObjectId();
+        const expectedResult = {
+          matchedCount: 1,
+          modifiedCount: 1,
+          upsertedCount: 1,
+          upsertedId: oid,
+          result: { ok: 1, n: 1, nModified: 1 },
+          connection: null,
+          acknowledged: true
+        } as any;
+        serviceProvider.updateOne.resolves(expectedResult);
+        const result = await shard.enableAutoMerger('ns');
+        expect(result).to.deep.equal(new UpdateResult(true, 1, 1, 1, oid));
+      });
+
+      it('throws if serviceProvider.updateOne rejects', async() => {
+        serviceProvider.runCommandWithCheck.resolves({ ok: 1, msg: 'isdbgrid' });
+        const expectedError = new Error();
+        serviceProvider.updateOne.rejects(expectedError);
+        const caughtError = await shard.disableBalancing('ns')
+          .catch(e => e);
+        expect(caughtError).to.equal(expectedError);
+      });
+
+      it('throws if not mongos', async() => {
+        const expectedResult = { ok: 1 } as any;
+        serviceProvider.runCommandWithCheck.resolves({ ok: 1, msg: 'not dbgrid' });
+        serviceProvider.updateOne.resolves(expectedResult);
+        await shard.enableAutoMerger('ns');
+        expect(warnSpy.calledOnce).to.equal(true);
+      });
+    });
```

Comment:
To be clear, what I meant on Slack was that I had assumed that there were tests in the `integration` section below :)

Reply:
Will check. I thought that I copied all the code for the things I copy/pasted from :)